aspects/aspects_test.go

```diff
@@ -270,3 +270,116 @@ func (s *aspectSuite) TestAspectsAccessControl(c *C) {
 		}
 	}
 }
+
+type spyDataBag struct {
```

Comment:
in the code base so far we tend to use witness instead of spy for this kind of test pattern